src/tools/recvfuzz/recvfuzz.cpp

```diff
@@ -505,25 +633,225 @@ void buildInitialPacket(CXPLAT_SOCKET* Binding, CXPLAT_ROUTE Route, int64_t* Pac
     }
 }
 
-void fuzzHandshakePacket() {
-// TODO
-}
-
 void fuzz(CXPLAT_SOCKET* Binding, CXPLAT_ROUTE Route) {
     int64_t PacketCount = 0;
     int64_t TotalByteCount = 0;
     uint8_t mode;
     uint64_t StartTimeMs = CxPlatTimeMs64();
+    uint64_t SrcCid;
+    PacketParams HandshakePacketParams = {
+        sizeof(uint64_t),
+        sizeof(uint64_t),
+        nullptr,
+        nullptr,
+        0,
+        1
+    };
+    CxPlatEventInitialize(&(RecvPacketEvent), TRUE, FALSE);
+    HandshakePacketParams.FrameTypes[0] = QUIC_FRAME_CRYPTO;
+    TlsContext HandshakeClientContext;
+    HandshakeClientContext.CreateContext();
+    TlsContext InitialClientContext;
+    InitialClientContext.CreateContext();
+    bool ServerHello = FALSE;
+    uint8_t recvBuffer[8192];
+    uint32_t bufferoffset = 0;
+    bool handshakeComplete = FALSE;
+    bool indicateMode = TRUE;
     while (CxPlatTimeDiff64(StartTimeMs, CxPlatTimeMs64()) < RunTimeMs) {
-        mode = 0; //(uint8_t)GetRandom(2);
+        mode = (uint8_t)GetRandom(2);
         if (mode == 0) {
-            buildInitialPacket(Binding, Route, &PacketCount, &TotalByteCount);
+            PacketParams InitialPacketParams = {
+                sizeof(uint64_t),
+                sizeof(uint64_t),
+                nullptr,
+                nullptr,
+                0,
+                1
+            };
+            if (indicateMode == TRUE) {
+                printf("Sending Initial Packets\n");
+                indicateMode = FALSE;
+            }
+            InitialPacketParams.PacketType = QUIC_INITIAL_V1;
+            InitialPacketParams.FrameTypes[0] = QUIC_FRAME_CRYPTO;
+            InitialPacketParams.mode = 0;
+            sendPacket(Binding, Route, &PacketCount, &TotalByteCount, &InitialPacketParams, true, &InitialClientContext);
         } else if (mode == 1) {
-            fuzzHandshakePacket();
+            if (handshakeComplete == FALSE) {
+                CxPlatEventInitialize(&(RecvPacketEvent), TRUE, FALSE);
```

Comment:
Not correct usage. Since this is a CxPlatEvent, the C++ wrappers handle this stuff. Please double check all usage of this.